lib/timers.js

```diff
@@ -483,152 +577,3 @@ exports.clearImmediate = function(immediate) {
     process._needImmediateCallback = false;
   }
 };
-
-
-// Internal APIs that need timeouts should use timers._unrefActive instead of
-// timers.active as internal timeouts shouldn't hold the loop open
-
-var unrefList, unrefTimer;
-
-function _makeTimerTimeout(timer) {
-  var domain = timer.domain;
-  var msecs = timer._idleTimeout;
-
-  L.remove(timer);
-
-  // Timer has been unenrolled by another timer that fired at the same time,
```

Comment:
Ok, after some meticulous looking over I've determined this check is not necessary.

This was required because previously `unenroll` had no way of actually removing an internal timer from the internal list, so instead we had the internals implementation check for the zeroed timeout duration. (Which I think was not very good practice, but that doesn't really matter anymore.)